apps/lend/src/hooks/usePageOnMount.ts

```diff
@@ -263,3 +290,24 @@ function getOptions(key: ConnectState['stage'], options: ConnectState['options']
   }
   return options
 }
+
+async function _isNetworkSwitched(newChainId: number) {
+  try {
+    if (window?.ethereum) {
+      const requestChainIdHex = await window.ethereum.request({ method: 'eth_chainId', params: [] })
```

Comment:
Don't use low-level calls, use ethers.
```suggestion
      const provider = new BrowserProvider(window.ethereum);
      const { chainId } = await provider.getNetwork()
```

Reply:
Or even better to use `wagmi` or `viem` from the onboard package
```ts
import wagmi from '@web3-onboard/wagmi'
import {
  sendTransaction as wagmiSendTransaction,
  switchChain,
  disconnect,
  getConnectors
} from '@web3-onboard/wagmi'
import { parseEther, isHex, fromHex } from 'viem'
```